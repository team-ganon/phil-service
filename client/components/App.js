import React, { Component } from 'react';
import reviewApiCall from './api';
import style from '../CSS/app.css';
import ReviewsNav from './ReviewsNav';
import ReviewsHeader from './ReviewsHeader';
import Review from './Review';
import Rating from './Rating';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: [],
      currentPlace: 0,
      currentReviews: [],
      total: 0,
      average: 0,
      error: false,
      cleanliness: 0,
      communication: 0,
      value: 0,
      accuracy: 0,
      checkIn: 0,
      location: 0
    };

    this.currentForward = () => {
      let current = this.state.reviews.slice(this.state.currentPlace + 7, this.state.currentPlace + 14);
      this.setState({
        currentPlace: this.state.currentPlace + 7,
        currentReviews: current
      });
    };

    this.currentBackword = () => {
      let current = this.state.reviews.slice(this.state.currentPlace - 7, this.state.currentPlace);
      this.setState({
        currentPlace: this.state.currentPlace - 7,
        currentReviews: current
      });
    };
  }

  componentDidMount() {
    let endPoint = window.location.href.split('=');
    reviewApiCall(endPoint[1])
      .then(result => console.log(result))
      .catch(e => this.setState(e));
  }

  render() {
    return (
      <div className={style.app}>
        <ReviewsHeader 
          average={this.state.average}
          total={this.state.total}
        />
        <div className={style.ratingHolderDiv}>
          <Rating name={{name: 'Cleanliness'}}score={this.state.cleanliness}/>
          <Rating name={{name: 'Communication'}}score={this.state.communication}/>
          <Rating name={{name: 'Value'}}score={this.state.value}/>
          <Rating name={{name: 'Accuracy'}}score={this.state.accuracy}/>
          <Rating name={{name: 'Check-in'}}score={this.state.checkIn}/>
          <Rating name={{name: 'Location'}}score={this.state.location}/>
        </div>
        {this.state.currentReviews.map((review, i) => <Review review={review} key={i} />)}
        {this.state.reviews.length > 0 && 
          <ReviewsNav 
            currentPlace={this.state.currentPlace}
            currentForward={this.currentForward}
            currentBackword={this.currentBackword}
            reviews={this.state.reviews}
          />
        }
      </div>
    );
  }
}

export default App;