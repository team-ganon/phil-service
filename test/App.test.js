import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../client/components/App';

describe('App Component', () => {

  it('<App /> currentPlace state should be zero', () => {
    const wrapper = shallow(<App />);
    const currentPlaceState = wrapper.state().currentPlace;
    expect(currentPlaceState).toEqual(0);
  });

  it('App SnapShot Test', () => {
    const AppJson = renderer.create(<App />).toJSON();
    expect(AppJson).toMatchSnapshot();
  });
  
});