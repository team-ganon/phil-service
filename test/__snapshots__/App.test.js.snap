// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App SnapShot Test 1`] = `
<div
  className="app"
>
  <div
    className="reviewHeaderDiv"
  >
    <h2
      className="reviewsHeaderH2"
    >
      Reviews
    </h2>
    <div
      className="bottomDivOfHeader"
    >
      <div
        className="containerForStarReviews"
      >
        <div
          className="star"
        >
          <ion-icon
            name="star"
          />
          <p
            className="averageTotal"
          >
            0
          </p>
        </div>
        <p
          className="average"
        >
          <span
            className="reviewCount"
          >
            0
          </span>
           reviews
        </p>
      </div>
      <div
        className="search"
      >
        <input
          placeholder="Search Reviews"
          type="text"
        />
        <ion-icon
          name="search-outline"
        />
      </div>
    </div>
  </div>
  <div
    className="ratingHolderDiv"
  >
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Cleanliness
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Communication
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Value
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Accuracy
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Check-in
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
    <div
      className="ratingDiv"
    >
      <p
        className="title"
      >
        Location
      </p>
      <div
        className="ratingBarAndScore"
      >
        <div
          className="outerLineDiv"
        >
          <div
            className="innerLineDiv"
            style={
              Object {
                "width": "00%",
              }
            }
          />
        </div>
        <p
          className="score"
        >
          0
        </p>
      </div>
    </div>
  </div>
</div>
`;
